crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -171,12 +173,26 @@ impl ChunkCast for Utf8Chunked {
     fn cast(&self, data_type: &DataType) -> PolarsResult<Series> {
         match data_type {
             #[cfg(feature = "dtype-categorical")]
-            DataType::Categorical(_) => {
-                let iter = self.into_iter();
-                let mut builder = CategoricalChunkedBuilder::new(self.name(), self.len());
-                builder.drain_iter(iter);
-                let ca = builder.finish();
-                Ok(ca.into_series())
+            DataType::Categorical(rev_map) => match rev_map {
+                None => {
+                    let iter = self.into_iter();
+                    let mut builder = CategoricalChunkedBuilder::new(self.name(), self.len());
+                    builder.drain_iter(iter);
+                    let ca = builder.finish();
+                    Ok(ca.into_series())
+                },
+                Some(rev_map) => {
+                    polars_ensure!(!using_string_cache(), InvalidOperation: "Can not cast to categorical with fixed categories while global string cache is active");
```

Comment:
This might be a little harsh / unnecessary. 

The confusion comes when you are in the string cache and create a categorical with fixed categories which would make it a local and not a global.

Reply:
This should be possible. 

1. If `rev_map` is None and `using_string_cache`, the user casted to a `pl.Categorical`
2. if `rev_map` is `Some`, the only valid cast is `pl.Categorical(enum=[..])`.

I do think we should adapt `rev_map` to 

```rust
#[derive(Clone)]
pub enum RevMapping {
    /// Hashmap: maps the indexes from the global cache/categorical array to indexes in the local Utf8Array
    /// Utf8Array: caches the string values
    Global(PlHashMap<u32, u32>, Utf8Array<i64>, u32),
    /// Utf8Array: caches the string values
    Local(Utf8Array<i64>),
    UserDefined{
        // User defined categories
        categories: Utf8Array<i64>,
        // Hash to quickly check if categories are equal
        hash: u128
      
    }
}
```

Where `Global`/`Local` have the same constraints we currently have, and `UserDefined` can be compared with each other by indexes if the hashes are equal.